cbits/cbits.c

```diff
@@ -0,0 +1,56 @@
+#include <stdint.h>
+#include <string.h>
+#include <limits.h>
+#include <assert.h>
+
+/* Taken from:
+ *
+ * <http://stackoverflow.com/questions/776508/best-practices-for-circular-shift-rotate-operations-in-c>
+ *
+ * Intel is little-endian, we have to rotate right
```

Comment:
Yep, but the mask is defined actually as a bytestring, so if you 'rotate' on a little endian, you rotate the bytestring in opposite direction than on big endian.